biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/ChemCompGroupFactory.java

```diff
@@ -79,6 +77,8 @@ public static ChemComp getChemComp(String recordName){
 	public static void setChemCompProvider(ChemCompProvider provider) {
 		logger.debug("Setting new chem comp provider to "+provider.getClass().getCanonicalName());
 		chemCompProvider = provider;
+		// clear cache
+		cache.clear();
```

Comment:
Are there any potential problems here? I know there was discussion that we should do this dating back to before github, so I was very surprised that it hadn't been done.